packages/digitalproperty-network/lib/DigitalLandTitle.js

```diff
@@ -27,8 +27,27 @@ function onRegisterPropertyForSale(propertyForSale) {
     console.log('### onRegisterPropertyForSale ' + propertyForSale.toString());
     propertyForSale.title.forSale = true;
 
-    return getAssetRegistry('net.biz.digitalPropertyNetwork.LandTitle').then(function(result) {
-        return result.update(propertyForSale.title);
-    }
-    );
-}
+    var SID;
+    var salesRegistry;
+    
+    SID = propertyForSale.seller.personId + propertyForSale.title.titleId;
+    
+    console.log('###' + 'SID =' + SID);
+
+    return getAssetRegistry('net.biz.digitalPropertyNetwork.SalesAgreement')
+        .then(function (result) {
+            salesRegistry = result;
```

Comment:
salesRegistry looks to only be used in the code block of the following then(). It could just be returned as an input argument to the then() block that uses it rather than assigned to a var scoped to the enclosing function.